index.go

```diff
@@ -135,6 +135,20 @@ func (i *Index) ColumnLabel() string {
 	return v
 }
 
+// Options returns all options for this index.
+func (i *Index) Options() IndexOptions {
```

Comment:
We should rebase this on master and make this an `RLock`

Reply:
@jaffee that read lock was already on master. do you have any other requests for this PR?